File: PHP/Spectacle.php
<?php
require_once 'Database.php';

class Spectacle {
    private $id;
    private $titre;
    private $lieu;
    private $affiche;
    private $tags;
    private $duree;
    private $description_courte;
    private $description_longue;
    private $langue;
    private $age_minimum;
    private $photos;

    public function __construct($titre, $lieu, $affiche, $tags, $duree, $description_courte, $description_longue, $langue, $age_minimum, $photos, $id = null) {
        $this->id = $id;
        $this->titre = $titre;
        $this->lieu = $lieu;
        $this->affiche = $affiche;
        $this->tags = $tags;
        $this->duree = $duree;
        $this->description_courte = $description_courte;
        $this->description_longue = $description_longue;
        $this->langue = $langue;
        $this->age_minimum = $age_minimum;
        $this->photos = $photos;
    }

    public function getId() { return $this->id; }
    public function getTitre() { return $this->titre; }
    public function getLieu() { return $this->lieu; }
    public function getAffiche() { return $this->affiche; }
    public function getTags() { return $this->tags; }
    public function getDuree() { return $this->duree; }
    public function getDescriptionCourte() { return $this->description_courte; }
    public function getLangue() { return $this->langue; }
    public function getAgeMinimum() { return $this->age_minimum; }

    public static function getAll() {
        $pdo = Database::getConnection();
        $stmt = $pdo->query("SELECT * FROM Spectacle ORDER BY titre ASC");
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    public static function getById($id) {
        $pdo = Database::getConnection();
        $stmt = $pdo->prepare("SELECT * FROM Spectacle WHERE id = ?");
        $stmt->execute([$id]);
        return $stmt->fetch(PDO::FETCH_ASSOC);
    }

    public function save() {
        $pdo = Database::getConnection();
        $stmt = $pdo->prepare("
            INSERT INTO Spectacle (titre, lieu, affiche, tags, duree, description_courte, description_longue, langue, age_minimum, photos)
            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
        ");
        $stmt->execute([
            $this->titre,
            $this->lieu,
            $this->affiche,
            $this->tags,
            $this->duree,
            $this->description_courte,
            $this->description_longue,
            $this->langue,
            $this->age_minimum,
            $this->photos
        ]);
        $this->id = $pdo->lastInsertId();
    }

    public static function delete($id) {
        $pdo = Database::getConnection();
        $stmt = $pdo->prepare("DELETE FROM Spectacle WHERE id = ?");
        $stmt->execute([$id]);
    }

}
?>

File: src/main/java/fr/billetterie/model/Spectacle.java
package fr.billetterie.model;

public class Spectacle {

    private int id;
    private String titre;
    private String lieu;
    private String affiche;
    private String tags;
    private Integer duree;
    private String descriptionCourte;
    private String descriptionLongue;
    private String langue;
    private Integer ageMinimum;
    private String photos;

    public Spectacle() {
    }

    public Spectacle(String titre, String lieu, String affiche, String tags, Integer duree,
                     String descriptionCourte, String descriptionLongue, String langue,
                     Integer ageMinimum, String photos) {
        this(0, titre, lieu, affiche, tags, duree, descriptionCourte, descriptionLongue, langue, ageMinimum, photos);
    }

    public Spectacle(int id, String titre, String lieu, String affiche, String tags, Integer duree,
                     String descriptionCourte, String descriptionLongue, String langue,
                     Integer ageMinimum, String photos) {
        this.id = id;
        this.titre = titre;
        this.lieu = lieu;
        this.affiche = affiche;
        this.tags = tags;
        this.duree = duree;
        this.descriptionCourte = descriptionCourte;
        this.descriptionLongue = descriptionLongue;
        this.langue = langue;
        this.ageMinimum = ageMinimum;
        this.photos = photos;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public String getAffiche() {
        return affiche;
    }

    public void setAffiche(String affiche) {
        this.affiche = affiche;
    }

    public String getTags() {
        return tags;
    }

    public void setTags(String tags) {
        this.tags = tags;
    }

    public Integer getDuree() {
        return duree;
    }

    public void setDuree(Integer duree) {
        this.duree = duree;
    }

    public String getDescriptionCourte() {
        return descriptionCourte;
    }

    public void setDescriptionCourte(String descriptionCourte) {
        this.descriptionCourte = descriptionCourte;
    }

    public String getDescriptionLongue() {
        return descriptionLongue;
    }

    public void setDescriptionLongue(String descriptionLongue) {
        this.descriptionLongue = descriptionLongue;
    }

    public String getLangue() {
        return langue;
    }

    public void setLangue(String langue) {
        this.langue = langue;
    }

    public Integer getAgeMinimum() {
        return ageMinimum;
    }

    public void setAgeMinimum(Integer ageMinimum) {
        this.ageMinimum = ageMinimum;
    }

    public String getPhotos() {
        return photos;
    }

    public void setPhotos(String photos) {
        this.photos = photos;
    }

    public String getDescription() {
        return descriptionCourte;
    }

    public void setDescription(String description) {
        this.descriptionCourte = description;
    }

    @Override
    public String toString() {
        return titre + " (" + lieu + ")";
    }
}
